server.js

```diff
@@ -0,0 +1,34 @@
+
+const HyperswarmProxyWSServer = require('hyperswarm-proxy-ws/server')
+const { SignalServer } = require('@geut/discovery-swarm-webrtc/server')
+const websocket = require('websocket-stream')
+
+const url = require('url')
+
+class HyperswarmServer extends HyperswarmProxyWSServer {
+  listenOnServer (server) {
+    this.server = server
+
+    const proxyWss = this.websocketServer = websocket.createServer({ noServer: true }, (socket) => {
+      this.handleStream(socket)
+    })
+
+    const signalServer = new SignalServer({ noServer: true })
+    const signalWss = signalServer.ws
+
+    server.on('upgrade', function upgrade(request, socket, head) {
+      const pathname = url.parse(request.url).pathname;
+
+      if (pathname === '/proxy') {
+        proxyWss.handleUpgrade(request, socket, head, (ws) => proxyWss.emit('connection', ws, request));
+      } else if (pathname === '/signal') {
+        signalWss.handleUpgrade(request, socket, head, (ws) => signalWss.emit('connection', ws, request));
+      } else {
+        console.warn('Invalid path. The path must be "/proxy" or "/signal"')
```

Comment:
Any chance we could default to the proxy if the path isn't set to signal? That way the old servers could be updated without breaking old clients.

Reply:
Yeah, that's a good idea!

Reply:
done